feat(AutoBlock):Better auto block

AnyOpenGL opened the pull request:
make autoblock tickOn and tickOff into random.

Commit:
feat(AutoBlock):random tickOn and tickOff

Commit:
fix(AutoBlock):random tickOn and tickOff

sqlerrorthing commented on src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/killaura/features/KillAuraAutoBlock.kt:
```diff
@@ -55,8 +55,13 @@ object KillAuraAutoBlock : ToggleableConfigurable(ModuleKillAura, "AutoBlocking"
-    val tickOn by int("TickOn", 0, 0..2, "ticks")
+    val tickOffRange by intRange("TickOffRange", 1..1, 0..5)
+    val tickOnRange by intRange("TickOnRange", 1..1, 0..5)
+    var tickOff : Int = 0
+        get() = tickOffRange.random()
+    var tickOn : Int = 0
+        get() = tickOnRange.random()
```
```suggestion
    val tickOff get() = tickOffRange.random()
    val tickOn get() = tickOnRange.random()
```

1zun4 committed:
Update src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/killaura/features/KillAuraAutoBlock.kt

Co-authored-by: sqlerrorthing <148702857+sqlerrorthing@users.noreply.github.com>

Commit:
optimized(AutoBlock):optimized some values.

Commit:
Merge remote-tracking branch 'origin/feat/BetterAutoBlock' into feat/BetterAutoBlock

# Conflicts:
#	src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/combat/killaura/features/KillAuraAutoBlock.kt

AnyOpenGL commented:
I viewed.

1zun4 committed:
Merge branch 'nextgen' into fork/AnyOpenGL/feat/BetterAutoBlock

1zun4 committed:
fix: bad random pattern

Without a fixed current value, we will end up not knowing the current tick off and on values, resulting in problems. Also, if not used together with waitTicks(...) where this approach is fine, we will always end up with min-range.

1zun4 commented:
@AnyOpenGL Please read my commit of: https://github.com/CCBlueX/LiquidBounce/pull/5870/commits/f75689a961de1913c4a62c9ef4a946d23d67e4d9

This is the reason why I didn't merge this as it is now, because randomizing on every call will always result in min-range having a higher chance when randomizing every tick. Also, we will not have one consistent value for a game tick, but several, as each call will result in a different number than before. For example, our randomization returns 1 and we think: Ok cool, we are going to block! Then, all of a sudden, we get a 0 in the code and the if check we did before has been bypassed and will theoretically result in a bug.
